In [1]:
import numpy as np
import matplotlib.pyplot as plt
import statsmodels
import statsmodels.formula.api as smf

In [2]:
import seaborn as sns

iris = sns.load_dataset('iris')
iris.head(3)

sepal_length  sepal_width  petal_length  petal_width species
0           5.1          3.5           1.4          0.2  setosa
1           4.9          3.0           1.4          0.2  setosa
2           4.7          3.2           1.3          0.2  setosa

In [5]:
#
formula =  "sepal_length ~ sepal_width" #반응/종속변수 length, 설명/독립변수 width
data = iris
model = smf.ols(formula, data).fit() 
model

In [8]:
model.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:           sepal_length   R-squared:                       0.014
Model:                            OLS   Adj. R-squared:                  0.007
Method:                 Least Squares   F-statistic:                     2.074
Date:                Mon, 15 Jan 2024   Prob (F-statistic):              0.152
Time:                        04:05:42   Log-Likelihood:                -183.00
No. Observations:                 150   AIC:                             370.0
Df Residuals:                     148   BIC:                             376.0
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
===============================================================================
                  coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------
Intercept       6.5262      0.479     13.628      0.000       5.580       7.473
sepal_width    -0.2234      0.155     -1.440      0.152      -0.530       0.083
==============================================================================
Omnibus:                        4.389   Durbin-Watson:                   0.952
Prob(Omnibus):                  0.111   Jarque-Bera (JB):                4.237
Skew:                           0.360   Prob(JB):                        0.120
Kurtosis:                       2.600   Cond. No.                         24.2
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

#### 해설
- Adj. R-squared: 수정된 결정계수 -> 수정결정계수가 1에 가까울수록 이상적인 모델이다. 해당 모델은 별로임.
- coef: 회귀계수
- Intercept의 coef: 상수항, 절편
- sepal_width의 coef: 기울기
- P>|t|: t 검정통계량에 대한 p 값
    + width의 p값은 0.152이므로 0.05보다 크다. 귀무가설을 채택한다.
    + 회귀분석에서 귀무가설은 <기울기가 0이다.>
    + 기울기가 0이다 -> 종속변수와 독립변수는 관계가 없다.